File: apprefurbished/src/main/java/com/androidcollider/easyfin/managers/import_export_db/ImportExportDbModule.java
package com.androidcollider.easyfin.managers.import_export_db;

import android.content.Context;

import com.androidcollider.easyfin.managers.ui.toast.ToastManager;
import com.androidcollider.easyfin.repository.database.DbHelper;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * @author Ihor Bilous
 */

@Module
public class ImportExportDbModule {

    @Provides
    @Singleton
    ImportExportDbManager provideImportDbManager(Context context, DbHelper dbHelper, ToastManager toastManager) {
        return new ImportExportDbManager(context, dbHelper, toastManager);
    }
}

File: apprefurbished/src/main/java/com/androidcollider/easyfin/common/managers/import_export_db/ImportExportDbModule.kt
package com.androidcollider.easyfin.common.managers.import_export_db

import android.content.Context
import com.androidcollider.easyfin.common.managers.ui.toast.ToastManager
import com.androidcollider.easyfin.common.repository.database.DbHelper
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * @author Ihor Bilous
 */
@Module
class ImportExportDbModule {
    @Provides
    @Singleton
    fun provideImportDbManager(
        context: Context,
        dbHelper: DbHelper,
        toastManager: ToastManager
    ): ImportExportDbManager {
        return ImportExportDbManager(context, dbHelper, toastManager)
    }
}